packages/multichain/src/scope/types.ts

```diff
@@ -54,11 +72,15 @@ export type InternalScopeObject = {
 /**
  * Represents a keyed `scopeObject` as defined in
  * [CAIP-217](https://chainagnostic.org/CAIPs/caip-217), with the exception that
- * `scopeObject`s do not contain `references` in our internal representations of CAIP-25 session scopes.
- * e.g. We flatten each reference into its own scopeObject before storing them in a `endowment:caip25` permission.
+ * we resolve the `references` property into a scopeObject per reference and
+ * assign an empty array to the `accounts` property if not already defined
+ * to more easily read chain specific permissions.
```

Comment:
Feels like we should call out how/why this type is used since we have 3 of them now 🤣 .

- return type for `getSession` 
- type used for validation pre-serialization?

Reply:
https://github.com/MetaMask/core/pull/4961/commits/e6722625c08f844a12687d2d5912cb7e4d7e2fe7 maybe?